crypto/pkg/tls/tls.go

```diff
@@ -46,22 +45,91 @@ var tls13Ciphers = []uint16{
-func NewTLSConfig() *tls.Config {
+// Legacy Ciphers we want to support for backwards compatibility.
+var legacyCiphers = []uint16{
+	tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+}
+
+// DefaultCiphers returns the ciphers supported by TLS 1.3 and the PFS ciphers supported by TLS 1.2.
+func DefaultCiphers() []uint16 {
+	return append(tls13Ciphers, tls12Ciphers...)
+}
+
+func supportedCipherMap() map[string]uint16 {
+	cipherMap := make(map[string]uint16)
+	addCiphers := func(ciphers []uint16) {
+		for _, cipher := range ciphers {
+			cipherName := tls.CipherSuiteName(cipher)
+			cipherMap[cipherName] = cipher
+		}
+	}
+
+	addCiphers(DefaultCiphers())
+	addCiphers(legacyCiphers)
+
+	return cipherMap
+}
+
+// ParseTLSCiphers takes a comma-separated string of cipher names and returns a slice of uint16 representing the ciphers.
+// If ciphers is empty, it returns the default ciphers.
+// It returns an error if any of the cipher names are not supported.
+func ParseTLSCiphers(ciphers string) ([]uint16, error) {
+	if ciphers == "" {
+		return DefaultCiphers(), nil
+	}
+
+	var result []uint16
+	supportedCiphers := supportedCipherMap()
+
+	cipherNames := strings.Split(ciphers, ",")
+	for _, name := range cipherNames {
+		name = strings.TrimSpace(name)
+		cipherValue, ok := supportedCiphers[name]
+		if !ok {
+			return nil, fmt.Errorf("unsupported cipher: %s", name)
+		}
+		result = append(result, cipherValue)
+	}
+
+	return result, nil
+}
+
+// NewTLSConfig returns a tls.Config with the recommended default settings for Calico components, using the TLS cipher suite IDs provided in the cipherSuites parameter.
+// Based on build flags, boringCrypto may be used and fips strict mode may be enforced, which can override the parameters defined in this func.
+func NewTLSConfig(cipherSuites []uint16) *tls.Config {
 	log.WithField("BuiltWithBoringCrypto", BuiltWithBoringCrypto).Debug("creating a TLS config")
+	if cipherSuites == nil {
+		cipherSuites = DefaultCiphers()
+	}
 	return &tls.Config{
 		MinVersion:   tls.VersionTLS12,
 		MaxVersion:   tls.VersionTLS13,
-		CipherSuites: append(tls12Ciphers, tls13Ciphers...),
+		CipherSuites: cipherSuites,
 	}
 }
 
-// NewMutualTLSConfig generates a tls.Config configured to enable mTLS with clients using the provided cert, key, and CA file paths.
-// If any of the files cannot be read, an error is returned.
-func NewMutualTLSConfig(cert, key, ca string) (*tls.Config, error) {
+// NewTLSConfigFromString returns a tls.Config with the recommended default settings for Calico components, using TLS cipher suite names provided as a comma-separated string.
+// The names are parsed into cipher suite IDs, and if parsing fails, an error is returned.
+// Based on build flags, boringCrypto may be used and fips strict mode may be enforced, which can override the parameters defined in this func.
+func NewTLSConfigFromString(cipherSuites string) (*tls.Config, error) {
```

Comment:
```suggestion
func NewTLSConfigFromCipherString(cipherSuites string) (*tls.Config, error) {
```

We don't support setting other fields from string, so better to be explicit.